src/main/java/io/vertx/ext/mongo/impl/PublisherAdapter.java

```diff
@@ -37,6 +37,7 @@ private enum State {
   private final int batchSize;
 
   private State state;
+  private int remainingItemsInPublisherBatch;
```

Comment:
For clarity, I'm going to rename this to `requestedNotReceived` and `inFlight` to `receivedNotDelivered` before merging.